src/renderer/utils/createStorage.ts

```diff
@@ -1,73 +1,210 @@
-// Helper to work with browser's local/session storage api
+// Helper for work with persistent local storage (default: window.localStorage)
+// TODO: write unit/integration tests
 
-export interface IStorageHelperOptions {
-  addKeyPrefix?: boolean;
-  useSession?: boolean; // use `sessionStorage` instead of `localStorage`
-}
+import { app, remote } from "electron";
+import path from "path";
+import { ensureDirSync, readJsonSync, writeJson } from "fs-extra";
+import type { CreateObservableOptions } from "mobx/lib/api/observable";
+import { action, comparer, observable, reaction, toJS, when } from "mobx";
+import produce, { Draft, setAutoFreeze } from "immer";
+import { isEqual, isFunction, isObject } from "lodash";
+
+setAutoFreeze(false); // allow to merge observables
 
-export function createStorage<T>(key: string, defaultValue?: T, options?: IStorageHelperOptions) {
+export function createStorage<T>(key: string, defaultValue?: T, options?: StorageHelperOptions<T>) {
   return new StorageHelper(key, defaultValue, options);
 }
 
-export class StorageHelper<T> {
-  static keyPrefix = "lens_";
+export interface StorageHelperOptions<T> extends StorageConfiguration<T> {
+  autoInit?: boolean; // default: true
+}
 
-  static defaultOptions: IStorageHelperOptions = {
-    addKeyPrefix: true,
-    useSession: false,
-  };
+export interface StorageConfiguration<T> {
+  storage?: StorageAdapter<T>;
+  observable?: CreateObservableOptions;
+}
 
-  constructor(protected key: string, protected defaultValue?: T, protected options?: IStorageHelperOptions) {
-    this.options = Object.assign({}, StorageHelper.defaultOptions, options);
+export interface StorageAdapter<T> {
+  [metadata: string]: any;
+  getItem(key: string): T; // import
+  setItem(key: string, value: T): void; // export
+  removeItem?(key: string): void; // if not provided setItem(key,undefined) will be used
+  onChange?(data: { key: string, value: T, oldValue?: T }): void;
+}
 
-    if (this.options.addKeyPrefix) {
-      this.key = StorageHelper.keyPrefix + key;
+export class StorageHelper<T> {
+  static defaultOptions: StorageHelperOptions<any> = {
+    autoInit: true,
+    get storage() {
+      return jsonFileSyncStorageAdapter;
+    },
+    observable: {
+      deep: true,
+      equals: comparer.shallow,
     }
-  }
+  };
+
+  private data = observable.box<T>();
```

Comment:
Given that we have default options, we should type them as fully supplied. Furthermore, there shouldn't be a need for an outside inspector (except in tests where we can cast to any anyway) to view the options after construction.
```suggestion

  private options: Required<StorageHelperOptions<T>>;
  private data = observable.box<T>();
```